### ISSUE TITLE
Figure out the preferred order of attributes

### ISSUE THREAD

**Issue by johno** [johnotander/prettify-html#3]

**Comment by TroyAlford** [comment]
I'm not sure we should alter the order of attributes in a doc, by default, as part of prettifying it. If this is a feature that's needed, though - maybe a good alternative to doing it by default would be to allow the caller to pass in a config? Something like:

```
prettifyHtml('<html>...</html>', {
  attributes: {
    a: ['href', 'class'],
  }
})
```

With this sort of API, we could simply check whether the `el.tagName` has a preferred ordering - and if it does, pull all of the specified items up to be first. It would also allow flexibility for someone parsing XML or JSX, where you might have odd, or non-standard attributes, such as: `className`, `readOnly`, or `MyCustomAttribute` - which may also have a case-sensitivity to them.

**Comment by johno** [comment]
Yeah, perhaps we don't want to change order by default, but provide the option to do so. Though, tbh, I'm not even sure if this is really a pre `1.0.0` consideration. Would prolly be best to first invest time in getting solid fixtures/tests to handle formatting edge cases and the like.
